frontend/alert/components/AlertForm/index.tsx

```diff
@@ -199,11 +224,15 @@ const AlertForm = ({
         // register all selected sections
         const promises: Array<Promise<Response>> = [];
         selectedCourses.forEach((section) => {
-            const promise = doAPIRequest("/api/alert/registrations/", "POST", {
-                section: section.section_id,
-                auto_resubscribe: autoResub === "true",
-            });
-            promises.push(promise);
+            isCourseOpen(section.section_id).then(isOpen => {
+                if (!isOpen) {
+                    const promise = doAPIRequest("/api/alert/registrations/", "POST", {
+                        section: section.section_id,
+                        auto_resubscribe: autoResub === "true",
+                    });
+                    promises.push(promise);
```

Comment:
I think we would want to put `const promise = isCourseOpen...` above and then `promises.push(promise);` below so that the `promises` array is definitely populated when we call `allSettled` below. Currently each promise is only added to the array once its `isCourseOpen` request returns, which causes a race condition between that and the interpreter reaching `Promise.allSettled`.

Reply:
Oh thats right, good catch!